common/readers.py

```diff
@@ -0,0 +1,79 @@
+import gzip
+import os
+from sickle.models import Record
+from lxml import etree
+
+import yaml
+
+import boto3
+
+from oarepo_runtime.datastreams.readers import BaseReader, StreamEntry
+from oarepo_oaipmh_harvester.readers.sickle import expand_datestamp
+
+class S3HarvestReader(BaseReader):
+    def __init__(self, *, source=None, base_path=None, oai_run=None, oai_harvester_id=None, manual=None, **kwargs):
+        super().__init__(source=source, base_path=base_path, **kwargs)
```

Comment:
Please use https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#shared-credentials-file to fetch profile credentials & endpoint config for boto3